ShopAppTests/App/Account/Controllers/AddressFormViewControllerSpec.swift

```diff
@@ -0,0 +1,327 @@
+                        expect(statePicker.customData).toEventually(equal(["State1"]))
+                    })
+                    .disposed(by: disposeBag)
+                
+                viewModelMock.makeNameOfStates()
+            }
+            
+            it("should select country correctly") {
+                countryPicker.pickerView.selectRow(0, inComponent: 0, animated: false)
+                
+                let toolbar = countryPicker.textField.inputAccessoryView as? UIToolbar
+                let doneButton = toolbar?.items?.filter({ $0.title == "Button.Done".localizable }).first
+                UIApplication.shared.sendAction(doneButton!.action!, to: doneButton!.target, from: nil, for: nil)
+                
+                countryPicker.textField.sendActions(for: .editingChanged)
+                expect(viewController.viewModel.countryText.value) == "Country1"
+            }
+            
+            it("should select state correctly") {
+                viewModelMock.makeNameOfStates()
+                statePicker.pickerView.selectRow(0, inComponent: 0, animated: false)
+                
+                let toolbar = statePicker.textField.inputAccessoryView as? UIToolbar
+                let doneButton = toolbar?.items?.filter({ $0.title == "Button.Done".localizable }).first
+                UIApplication.shared.sendAction(doneButton!.action!, to: doneButton!.target, from: nil, for: nil)
+                
+                statePicker.textField.sendActions(for: .editingChanged)
+                expect(viewController.viewModel.stateText.value) == "State1"
+            }
+        }
+        
+        describe("when submit button tapped") {
+            beforeEach {
+                initData(addressNeeded: false)
+            }
+            
+            it("should have correct submit button tap binding") {
+                submitButton.sendActions(for: .touchUpInside)
+                expect(viewModelMock.submitButtonDidPress) == true
+            }
+        }
+        
+        describe("when submit button state changed") {
+            beforeEach {
+                initData(addressNeeded: false)
+            }
+            
+            it("should return addressValid false") {
+                viewModelMock.addressValid.value = false
+                expect(submitButton.isEnabled) == false
+            }
+            
+            it("should return addressValid true") {
+                viewModelMock.addressValid.value = true
+                expect(submitButton.isEnabled) == true
+            }
+        }
+        
+        describe("when submit tapped") {
+            beforeEach {
+                initData(addressNeeded: true)
+            }
+            
+            it("should return correct address") {
+                viewModelMock.makeSubmitAction()
+                expect(delegateMock.filledAddress).toNot(beNil())
+            }
+        }
+        
+        func initData(addressNeeded: Bool) {
```

Comment:
This func should be private

Reply:
Agree, but we have error 'Attribute 'private' can only be used in a non-local scope'. If we want to move this method out of 'spec()' function, it also need to move all properties (like viewController, viewModel, etc)

Reply:
Ok, lets use your way